core/arch/arm/kernel/unwind_arm32.c

```diff
@@ -367,10 +367,14 @@ void print_stack(int level)
 {
 	struct unwind_state state;
 
-	memset(&state, 0, sizeof(state));
+	memset(state.registers, 0, sizeof(state.registers));
+	/* r7: Thumb-style frame pointer */
+	state.registers[7] = read_r7();
```

Comment:
for consistency, could you use a `R7` label defined next to `FP`, `SP` and friends above ?